### PR TITLE
tidb: config tidb server labels

### PR DESCRIPTION
**Pull request by glorv**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by glorv** [comment]
@Yisaer @qw4990 @Oreoxmt  PTAL

File: follower-read.md [hunk 1]
```diff
@@ -32,10 +32,17 @@ set [session | global] tidb_replica_read = '<目标值>';
...
-- 当设置为 `leader-and-follower` 时，读请求会在 leader 和 follower 之间负载均衡。
+- 当设置为 `leader-and-follower` 时，TiDB 可以选择任意副本来执行读取操作，此时读请求会在 leader 和 follower 之间负载均衡。
+- 当设置为 `closest-replicas` 时，TiDB 会优先选择分布在同一个区域的副本执行读取操作，对应的副本可以是 leader 或 follower, 如果同区域内没有副本分布，则会从 leader 执行读取。
+- 当设置为 `closest-adaptive` 时，如果根据一个读请求的预估返回结果的大小大于或等于变量 `tidb_adaptive_closest_read_threshold` 的值时，会优先选择分布在同一个区域的副本执行读取操作，否则选择 leader 副本进行处理。
+
+> **注意：**
+>
+> 1. `closest-replicas` 或 `closest-adaptive` 依赖集群的各个副本按照指定的设置分布在各个区域，请参考 [通过拓扑 label 进行副本调度](/schedule-replicas-by-topology-labels.md) 为 PD 配置 `location-labels` 并为 TiDB 和 TiKV 设置正确的 Labels。TiDB 依赖 Label `zone` 匹配位于同一个区域的 TiKV，因此请务必在 PD 的 `location-labels` 配置中包含 `zone` 并确保每个 TiDB 和 TiKV 节点的 Labels 配置中包含 `zone`。
+> 如果是使用 TiDB Operator 部署的集群，请参考 [配置副本跨区域分布](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/configure-a-tidb-cluster#%E6%95%B0%E6%8D%AE%E7%9A%84%E9%AB%98%E5%8F%AF%E7%94%A8) 进行配置。
+> 2. 当设置为 `closest-adaptive` 时，为了防止读流量在各个区域分布不均衡，TiDB 会动态检测当前在线的所有 TiDB 和 TiKV 的区域分布是否均衡，如果发现在某一个或区域内仅包含 TiDB 或 TiKV 的节点，则会强制使用 leader 读取。例如，如果某个区域的 TiDB 全部挂掉，此时其他在线的 TiDB 将会降级至 leader 读取，当此区域至少 1 个 TiDB 节点重新上线后，则会恢复优先从同区域的副本读取。
```
Comment (Oreoxmt) [inline follower-read.md:45]:
```suggestion
> **注意：**
>
> - `closest-replicas` 或 `closest-adaptive` 依赖集群的各个副本按照指定的设置分布在各个区域，请参考[通过拓扑 label 进行副本调度](/schedule-replicas-by-topology-labels.md)为 PD 配置 `location-labels` 并为 TiDB 和 TiKV 设置正确的 `labels`。TiDB 依赖 label `zone` 匹配位于同一个区域的 TiKV，因此请**务必**在 PD 的 `location-labels` 配置中包含 `zone` 并确保每个 TiDB 和 TiKV 节点的 labels 配置中包含 `zone`。如果是使用 TiDB Operator 部署的集群，请参考[配置副本跨区域分布](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/configure-a-tidb-cluster#%E6%95%B0%E6%8D%AE%E7%9A%84%E9%AB%98%E5%8F%AF%E7%94%A8)进行配置。
> - 当设置为 `closest-adaptive` 时，为了防止读流量在各个区域分布不均衡，TiDB 会动态检测当前在线的所有 TiDB 和 TiKV 的区域分布是否均衡。如果发现在某一个区域内仅包含 TiDB 或 TiKV 的节点，则会强制使用 leader 读取。例如，如果某个区域的 TiDB 全部挂掉，此时其他在线的 TiDB 将会降级至 leader 读取。当此区域至少 1 个 TiDB 节点重新上线后，则会恢复优先从同区域的副本读取。
```

**Comment by Oreoxmt** [comment]
@Yisaer @qw4990 PTAL

**Comment by Oreoxmt** [comment]
@Yisaer @qw4990 PTAL

**Review by Yisaer — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: follower-read.md [hunk 2]
```diff
@@ -32,10 +32,16 @@ set [session | global] tidb_replica_read = '<目标值>';
...
 该变量用于设置期待的数据读取方式。
 
-- 当设定为默认值 `leader` 或者空字符串时，TiDB 会维持原有行为方式，将所有的读取操作都发送给 leader 副本处理。
+- 当设置为默认值 `leader` 或者空字符串时，TiDB 会维持原有行为方式，将所有的读取操作都发送给 leader 副本处理。
 - 当设置为 `follower` 时，TiDB 会选择 Region 的 follower 副本完成所有的数据读取操作。
-- 当 `tidb_replica_read` 的值设为 `leader-and-follower` 时，TiDB 可以选择任意副本来执行读取操作。
-- 当设置为 `leader-and-follower` 时，读请求会在 leader 和 follower 之间负载均衡。
+- 当设置为 `leader-and-follower` 时，TiDB 可以选择任意副本来执行读取操作，此时读请求会在 leader 和 follower 之间负载均衡。
+- 当设置为 `closest-replicas` 时，TiDB 会优先选择分布在同一区域的副本执行读取操作，对应的副本可以是 leader 或 follower。如果同一区域内没有副本分布，则会从 leader 执行读取。
+- 当设置为 `closest-adaptive` 时，当一个读请求的预估返回结果大于或等于变量 [`tidb_adaptive_closest_read_threshold`](/system-variables.md#tidb_adaptive_closest_read_threshold-从-v630-版本开始引入) 的值时，TiDB 会优先选择分布在同一区域的副本执行读取操作，否则选择 leader 副本进行处理。
```
Comment (ran-huang) [inline follower-read.md:39]: 这部分应该写到 [tidb_replica_read](https://github.com/pingcap/docs-cn/blob/e8286bf800a4eabdb8c7c60b5a665e890fa22b64/system-variables.md#tidb_replica_read-%E4%BB%8E-v40-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) 里。此处只需要保留与 follower read 相关的值。

File: system-variables.md [hunk 3]
```diff
@@ -358,6 +358,13 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 - 默认值：（随系统）
 - 该变量显示首次引导启动 TiDB 时的系统时区。另请参阅 [`time_zone`](#time_zone)。
 
+### `tidb_adaptive_closest_read_threshold` <span class="version-mark">从 v6.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`4096`
+- 这个变量用于控制开启 TiDB [Follower Read](/follower-read.md)功能时，当 `tidb_replica_read` 设置为 `closest-adaptive` 时，优先将读请求发送至 TiDB server 所在区域副本的阈值。当读请求预估的返回结果的大小超过此阈值时，TiDB 会将读请求优先发送至同一区域的副本，否则会发送至 leader 副本。
```
Comment (ran-huang) [inline system-variables.md:366]:
```suggestion
- 这个变量用于控制当 `tidb_replica_read` 设置为 `closest-adaptive` 时，优先将读请求发送至 TiDB server 所在区域副本的阈值。当读请求预估的返回结果的大小超过此阈值时，TiDB 会将读请求优先发送至同一区域的副本，否则会发送至 leader 副本。
```

File: follower-read.md (hunk 2 above)
Comment (Oreoxmt) [reply follower-read.md:39]: Done https://github.com/pingcap/docs-cn/pull/11242/commits/7d09f8bd7abc94eb91b9427a9a467d07ffd69677

**Review by ran-huang — APPROVED** [review]

File: follower-read.md [hunk 4]
```diff
@@ -32,10 +32,16 @@ set [session | global] tidb_replica_read = '<目标值>';
...
+
+> **注意：**
+>
+> - 当设置为 `closest-replicas` 或 `closest-adaptive` 时，集群的各个副本应当按照指定的设置分布在各个区域。请参考[通过拓扑 label 进行副本调度](/schedule-replicas-by-topology-labels.md)为 PD 配置 `location-labels` 并为 TiDB 和 TiKV 设置正确的 `labels`。TiDB 依赖 `zone` 标签匹配位于同一区域的 TiKV，因此请**务必**在 PD 的 `location-labels` 配置中包含 `zone` 并确保每个 TiDB 和 TiKV 节点的 `labels` 配置中包含 `zone`。如果是使用 TiDB Operator 部署的集群，请参考[配置副本跨区域分布](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/configure-a-tidb-cluster#%E6%95%B0%E6%8D%AE%E7%9A%84%E9%AB%98%E5%8F%AF%E7%94%A8)进行配置。
```
Comment (ran-huang) [inline follower-read.md:43]: 第一句话是用户需要采取的行为？如果是，建议改成这样，否则读起来像在描述副本原本的状态。
```suggestion
> - 当设置为 `closest-replicas` 或 `closest-adaptive` 时，你需要配置集群以确保副本按照指定的设置分布在各个区域。请参考[通过拓扑 label 进行副本调度](/schedule-replicas-by-topology-labels.md)为 PD 配置 `location-labels` 并为 TiDB 和 TiKV 设置正确的 `labels`。TiDB 依赖 `zone` 标签匹配位于同一区域的 TiKV，因此请**务必**在 PD 的 `location-labels` 配置中包含 `zone` 并确保每个 TiDB 和 TiKV 节点的 `labels` 配置中包含 `zone`。如果是使用 TiDB Operator 部署的集群，请参考[配置副本跨区域分布](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/configure-a-tidb-cluster#%E6%95%B0%E6%8D%AE%E7%9A%84%E9%AB%98%E5%8F%AF%E7%94%A8)进行配置。
```

File: follower-read.md [hunk 5]
```diff
@@ -32,10 +32,16 @@ set [session | global] tidb_replica_read = '<目标值>';
...
+> - 当设置为 `closest-adaptive` 时，为了防止读流量在各个区域分布不均衡，TiDB 会动态检测当前在线的所有 TiDB 和 TiKV 的区域分布是否均衡。如果发现在某一个区域内仅包含 TiDB 或 TiKV 的节点，则会强制使用 leader 读取。例如，如果某个区域的 TiDB 全部无法连接，此时其他在线的 TiDB 将会降级至 leader 读取。当此区域有至少 1 个 TiDB 节点重新上线，则会恢复优先从同一区域的副本读取。
```
Comment (ran-huang) [inline follower-read.md:44]: 1. 这里两条写成**注意**反而可读性不太好。
2. 第一条是用户需要采取的行为，第二条是描述内部工作机制（不需要用户采取行为）。笼统地写在一个*注意*里可能导致用户遗漏信息。

File: system-variables.md [hunk 6]
```diff
@@ -1980,12 +1987,14 @@ explain select * from t where age=5;
...
+- 可选值：`leader`，`follower`，`leader-and-follower`，`closest-replicas`，`closest-adaptive`
+- 这个变量用于控制 TiDB 读取数据的位置，有以下几个选择：
+
+    * `leader`：只从 leader 节点读取。
+    * `follower`：只从 follower 节点读取。
+    * `leader-and-follower`：从 leader 或 follower 节点读取。
+    * `closest-replicas`：优先从分布在同一区域的 leader 或 follower 节点读取，如果同一区域内没有副本分布，则会从 leader 读取。
+    * `closest-adaptive`：当一个读请求的预估返回结果大于或等于变量 [`tidb_adaptive_closest_read_threshold`](/system-variables.md#tidb_adaptive_closest_read_threshold-从-v630-版本开始引入) 的值时，优先从分布在同一区域的 leader 或 follower 节点读取，否则从 leader 节点读取。
 
 更多细节，见 [Follower Read](/follower-read.md)。
```
Comment (ran-huang) [inline system-variables.md:1999]: 这里我感觉给到完整的可选值就可以了，具体的取值含义不用维护两份。

```suggestion
- 这个变量用于控制 TiDB 的 Follower Read 功能的行为。

关于使用方式与实现原理，见 [Follower Read](/follower-read.md)。
```

File: follower-read.md (hunk 5 above)
Comment (Oreoxmt) [reply follower-read.md]: 注意里只放了用户需要采取的行为，第二条放到了 `closest-adaptive` 可选值的描述里 https://github.com/pingcap/docs-cn/pull/11242/commits/92ffeb45bdcd101fdb12aa8123e4c7730e9471ca

**Review by ran-huang — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Pull request merged**